browser/net/brave_network_delegate_browsertest.cc

```diff
@@ -152,3 +194,23 @@ IN_PROC_BROWSER_TEST_F(BraveNetworkDelegateBrowserTest,
       content::GetCookies(browser()->profile(), GURL("http://a.com/"));
   EXPECT_TRUE(cookie.empty()) << "Actual cookie: " << cookie;
 }
+
+IN_PROC_BROWSER_TEST_F(BraveNetworkDelegateBrowserTest,
+                       IframeGoogleAccountsCookieAllowed) {
+  NavigateToPageWithFrame("a.com");
+  ExpectCookiesOnHost("accounts.google.com", "");
+
+  NavigateFrameTo("accounts.google.com", "/set-cookie?oauth=true");
+  ExpectCookiesOnHost("accounts.google.com", "oauth=true");
+}
+
+IN_PROC_BROWSER_TEST_F(BraveNetworkDelegateBrowserTest,
+                       IframeGoogleAccountsCookieBlocked) {
+  BlockGoogleOAuthCookies();
```

Comment:
we need a test, showing that the new preference doesn't override user choice in per-site shields settings

Reply:
I actually requested this in the last review and should verify your comment below

Reply:
Added here: https://github.com/brave/brave-core/pull/3784/files#diff-a3b1f2bffde11cef192a185a65955508R350